File: app/lib/models/relatorio.dart
/// Modelo de Relatorio de Evolucao (UC007).
class Relatorio {
  final String id;
  final String grupoId;
  final DateTime periodoInicio;
  final DateTime periodoFim;
  final List<String> categorias;
  final String? urlPdf;
  final int totalEntradas;
  final int versao;

  Relatorio({
    required this.id,
    required this.grupoId,
    required this.periodoInicio,
    required this.periodoFim,
    required this.categorias,
    required this.urlPdf,
    required this.totalEntradas,
    required this.versao,
  });

  factory Relatorio.fromJson(Map<String, dynamic> j) => Relatorio(
        id: j['id'],
        grupoId: j['grupoId'] ?? '',
        periodoInicio: DateTime.tryParse(j['periodoInicio']?.toString() ?? '') ?? DateTime.now(),
        periodoFim: DateTime.tryParse(j['periodoFim']?.toString() ?? '') ?? DateTime.now(),
        categorias: ((j['categorias'] ?? []) as List).map((e) => e.toString()).toList(),
        urlPdf: j['urlPdf'],
        totalEntradas: j['totalEntradas'] ?? 0,
        versao: j['versao'] ?? 1,
      );
}

File: backend/src/models/Relatorio.js
'use strict';

// Entidade Relatorio de Evolucao (<<entity>>) - UC007.
class Relatorio {
  constructor({
    id,
    grupoId,
    periodoInicio, // ISO date
    periodoFim, // ISO date
    categorias = [], // vazio = todas
    urlPdf = null,
    totalEntradas = 0,
    versao = 1,
    geradoPor,
    geradoEm = new Date(),
  }) {
    this.id = id;
    this.grupoId = grupoId;
    this.periodoInicio = periodoInicio;
    this.periodoFim = periodoFim;
    this.categorias = categorias;
    this.urlPdf = urlPdf;
    this.totalEntradas = totalEntradas;
    this.versao = versao;
    this.geradoPor = geradoPor;
    this.geradoEm = geradoEm;
  }

  toJSON() {
    return { ...this };
  }
}

module.exports = Relatorio;
